hack/dind-cluster.sh

```diff
@@ -170,6 +185,102 @@ cluster's rc file to configure the bash environment:
   fi
 }
 
+function get-registry-ip() {
+  local default_iface=$(route | grep default | awk '{print $8}')
+  echo $(ip addr show ${default_iface} | grep -Po 'inet \K[\d.]+')
+}
+
+function add-registry-to-runtime() {
+  local registry=$1
+  local runtime=$2
+
+  for cid in $( ${DOCKER_CMD} ps -qa --filter "name=${MASTER_NAME}|${NODE_PREFIX}" ); do
+    # Wait for dbus service so that we can restart runtime service
+    local count=60
+    while [[ "${count}" -gt "0" ]]; do
+      local error=0
+      ${DOCKER_CMD} exec -t ${cid} sh -c "systemctl is-active --quiet dbus" 2>&1 > /dev/null || error=1
+      if [[ "${error}" -eq "0" ]]; then
+        echo ""
+        break
+      fi
+      echo -n "."
+      sleep 1
+      ((count--))
+    done
+
+    if [[ "${runtime}" = "dockershim" ]]; then
+      ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i '/^\[registries.search\]$/{\$!{N;s/^\[registries.search\]\nregistries .*/\[registries.search\]\nregistries = \[\"${registry}\", \"docker.io\", \"registry.fedoraproject.org\", \"registry.access.redhat.com\"\]/g;ty;P;D;:y}}' /etc/containers/registries.conf"
+      ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i '/^\[registries.insecure\]$/{\$!{N;s/^\[registries.insecure\]\nregistries .*/\[registries.insecure\]\nregistries = \[\"${registry}\"\]/g;ty;P;D;:y}}' /etc/containers/registries.conf"
+      ${DOCKER_CMD} exec -t ${cid} sh -c "systemctl restart docker"
+    else
+      ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i 's/^registries .*/registries = [\n\t\"${registry}\",/g' /etc/crio/crio.conf"
+      ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i 's/^insecure_registries .*/insecure_registries = [\n\t\"${registry}\",/g' /etc/crio/crio.conf"
+      ${DOCKER_CMD} exec -t ${cid} sh -c "systemctl restart crio"
+    fi
+  done
+}
+
+function enable-local-registry() {
+  local origin_root=$1
+  local config_root=$2
+  local ip=$3
+  local images=($4)
+  local tag=$(pushd "${origin_root}" >/dev/null ; git describe --abbrev=0 ; popd > /dev/null)
+
+  # Start local registry
+  ${DOCKER_CMD} run -d -p ${REGISTRY_PORT}:5000 --restart=always --name ${REGISTRY_NAME} registry:2 > /dev/null
+  echo "Created local registry '${REGISTRY_NAME}' at ${ip}:${REGISTRY_PORT}"
+
+  # Build, tag and push local images
```

Comment:
Should there be a way to do this again later?  I think a separate command to add all images to the local registry would be handy.

Reply:
Yes, that will be useful.
I was thinking instead of a new command, piggybacking on 'refresh' command may be handy because most of the time we need updated openshift binary along with the image changes.

Reply:
maybe a hack/local_registry.sh that performs the registry tasks could be useful